java/runtime/src/main/java/io/ray/runtime/object/ObjectSerializer.java

```diff
@@ -106,8 +114,9 @@ public static NativeRayObject serialize(Object object) {
         buffer.get(bytes);
       }
       return new NativeRayObject(bytes, OBJECT_METADATA_TYPE_RAW);
-    } else if (object instanceof RayTaskException) {
-      byte[] serializedBytes = Serializer.encode(object).getLeft();
+    } else if (object instanceof NativeRayException) {
+      NativeRayException taskException = (NativeRayException) object;
+      byte[] serializedBytes = Serializer.encode(taskException.toBytes()).getLeft();
```

Comment:
Can you add a comment here that `serializedBytes` != `taskException.toBytes()`, and explain. I assume the actual data stored in `NativeRayObject.data` is a msgpacked protobuf data. Am I right?

Reply:
Yes, the `serializedBytes` != `taskException.toBytes()`. I will add more comments here.